Improve SUSE Manger 5.0 proxy installation and migration (#3158)
uyuni-project/uyuni-docs: modules/installation-and-upgrade/pages/container-deployment/generate_proxy_config.adoc

// Add section on how to generated the proxy configuration files

[[proxy-setup-containers-generate-config]]
== Generate the Proxy Configuration

The configuration archive of the {productname} Proxy is generated by the {productname} Server.
Each additional Proxy requires its own configuration archive.

[NOTE]
====

> 2 GB represents the default proxy squid cache size. This will need to be adjusted for your environment.

Reviewer: This note is without any context before, so it could be confusing for the user. Maybe, we should move it close to where we are describing squid cache size.

Author: This note in in the section about generating the config, and should apply to all available methods

Reviewer 2: In this case, we'd better repeat with all available methods.  Maybe, with an include statement as a snippet (we can do this later).

For the moment, we could add here at the beginning of the note, as the first sentence:

> "Keep in mind that before actually installing (deploying) the proxy you must probably adjust the Squid cache size."

====

[IMPORTANT]
====
For Podman deployment, the container host for the {productname} Proxy must be registered as a client to the {productname} Server prior to generating this proxy configuration.
====

> If a Proxy FQDN is used to generate Proxy container configuration that is not a registered minion (kubernetes use case), a new system entry will appear in system list. This new entry will be shown under previously entered Proxy FQDN value and will be of Foreign system type.

Reviewer 3: ```suggestion
If a Proxy FQDN is used to generate Proxy container configuration that is not a registered minion (kubernetes use case), a new system entry will appear in system list. 
This new entry will be shown under previously entered Proxy FQDN value and will be of Foreign system type.
```

// Add section on how to generated the proxy configuration files

[[proxy-setup-containers-generate-config]]

> == Generate the Proxy Configuration

Author: This section have 3 sub-section that are all viable alternatives to generated the proxy configuration file. I didn't mentioned that they are alternatives, and users should use just one. Do you think we should make it more explicit?

// Add section on how to generated the proxy configuration files

[[proxy-setup-containers-generate-config]]
== Generate the Proxy Configuration

The configuration archive of the {productname} Proxy is generated by the {productname} Server.
Each additional Proxy requires its own configuration archive.

[NOTE]
====
2 GB represents the default proxy squid cache size. This will need to be adjusted for your environment.
====

[IMPORTANT]
====
For Podman deployment, the container host for the {productname} Proxy must be registered as a client to the {productname} Server prior to generating this proxy configuration.
====

> If a Proxy FQDN is used to generate Proxy container configuration that is not a registered minion (kubernetes use case), a new system entry will appear in system list. 

Reviewer 2: ```suggestion
If a proxy FQDN is used to generate a proxy container configuration that is not a registered client (as in the Kubernetes use case), a new system entry will appear in system list. 
```

This new entry will be shown under previously entered Proxy FQDN value and will be of [literal]``Foreign`` system type.

=== Generate the Proxy Configuration with {webui}

[[proc-proxy-containers-setup-webui]]
.Procedure: Generating a Proxy Container Configuration using {webui}

. In the {webui}, navigate to menu:Systems[Proxy Configuration] and fill the required data:
. In the [guimenu]``Proxy FQDN`` field type fully qualified domain name for the proxy.
. In the [guimenu]``Parent FQDN`` field type fully qualified domain name for the {productname} Server or another {productname} Proxy.
. In the [guimenu]``Proxy SSH port`` field type SSH port on which SSH service is listening on {productname} Proxy. Recommended is to keep default 8022.
. In the [guimenu]``Max Squid cache size [MB]`` field type maximal allowed size for Squid cache.
  Recommended is to use at most 60% of available storage for the containers.
+

[NOTE]
====
2 GB represents the default proxy squid cache size.

> This will need to be adjusted for your environment.

Reviewer 2: ```suggestion
This will need to be adjusted for your environment.
====

```
